core/src/main/java/org/infinispan/remoting/rpc/RpcManagerImpl.java

```diff
@@ -106,24 +137,165 @@ public String getPendingViewAsString() {
       return pendingCH != null ? pendingCH.getMembers().toString() : "null";
    }
 
+   @Override
+   public <T> CompletionStage<T> invokeCommand(Address target, ReplicableCommand command,
+                                               ResponseCollector<T> collector, RpcOptions rpcOptions) {
+      checkTopologyId(command);
```

Comment:
Most of the method looks the same. can't we avoid that code duplication? 
```
invokeCommand(Address...)
invokeCommand(Collection<Address> targets...)
invokeCommandOnAll(...)
invokeCommandStaggered(...)
```

Reply:
I wanted to avoid creating another lambda per invocation on top of the updateStatistics one. I'll have to check if escape analysis can remove the allocation, otherwise I'd rather keep the duplication.